src/AccordionContainer/AccordionContainer.tsx

```diff
@@ -109,10 +112,22 @@ export class Provider extends React.Component<ProviderProps, ProviderState> {
             (state: ProviderState) => ({
                 items: state.items.map((item: Item) => {
                     if (item.uuid === key) {
-                        return {
-                            ...item,
-                            expanded,
-                        };
+                        if (
```

Comment:
Two notes on this little block:

- It could be moved above the `this.setState` block and just do `return;` to bail out. That way, if the conditions are met then `setState` is never called at all, and an unnecessary re-render is prevented.
- The `.length < 2` is slightly ambiguous. I think more correctly it would be `.length === 1`, because this logic should not apply if `length = 0` or if `length = 1.5`, for example. Both of those examples would not eventuate in practice (particularly 1.5 haha) but still doesn't hurt to be explicit.

Reply:
Ah yep, good thinking - just made those changes :)